scripts/devops_tasks/test_regression.py

```diff
@@ -0,0 +1,364 @@
+
+    def deactivate(self, dep_pkg_root_path):
+        # This function can be used to reset code repo to master branch and also to deactivate virtual env
+        # Revert to master branch
+        run_check_call(["git", "clean", "-fd"], dep_pkg_root_path)
+        run_check_call(["git", "checkout", GIT_MASTER_BRANCH], dep_pkg_root_path)
+        self.venv.deactivate()
+
+
+class RegressionTest:
+    def __init__(self, context, package_dependency_dict):
+        self.context = context
+        self.package_dependency_dict = package_dependency_dict
+
+    def run(self):
+        pkg_name = self.context.package_name
+        if pkg_name in self.package_dependency_dict:
+            dep_packages = self.package_dependency_dict[pkg_name]
+            logging.info("Running regression test for {}".format(pkg_name))
+            logging.info(
+                "Dependent packages for [{0}]: {1}".format(pkg_name, dep_packages)
+            )
+
+            for dep_pkg_path in dep_packages:
+                dep_pkg_name, _, _, _ = parse_setup(dep_pkg_path)
+                logging.info(
+                    "Starting regression test of {0} against released {1}".format(
+                        pkg_name, dep_pkg_name
+                    )
+                )
+                self._run_test(dep_pkg_path)
+                logging.info(
+                    "Completed regression test of {0} against released {1}".format(
+                        pkg_name, dep_pkg_name
+                    )
+                )
+            logging.info("Completed regression test for {}".format(pkg_name))
+        else:
+            logging.info(
+                "Package {} is not added as required by any package".format(pkg_name)
+            )
+
+    def _run_test(self, dep_pkg_path):
+        # find GA released tags for package and run test using that code base
+        dep_pkg_name, _, _, _ = parse_setup(dep_pkg_path)
+        release_tag = get_release_tag(dep_pkg_name, self.context.is_latest_depend_test)
+        if not release_tag:
+            logging.error(
+                "Release tag is not avaiable. Skipping package {} from test".format(
+                    dep_pkg_name
+                )
+            )
+            sys.exit(1)
+
+        # Ommit based on package name and release tag
+        pkg_releae_tag = "{0}:{1}".format(dep_pkg_name, release_tag)
+        if pkg_release_tag in PACKAGE_RELEASE_TAGS_TO_EXCLUDE:
+            logging.info("Package {0} and release tag {1} is excluded from regression test")
+            return
+
+        # Get code repo with released tag of dependent package
+        checkout_code_repo(release_tag, dep_pkg_path)
+
+        try:
+            # activate virtual env
+            self.context.activate(dep_pkg_path)
+            # install packages required to run tests after updating relative reference to abspath
+            run_check_call(
+                [
+                    self.context.python_symlink,
```

Comment:
Yep. I believe you can remove all `activate` calls and you'll get the same results.